bot/exts/halloween/candy_collection.py

```diff
@@ -27,13 +26,16 @@ class CandyCollection(commands.Cog):
 
     def __init__(self, bot: commands.Bot):
         self.bot = bot
+
         with open(json_location, encoding="utf8") as candy:
-            self.candy_json = json.load(candy)
-            self.msg_reacted = self.candy_json['msg_reacted']
-        self.get_candyinfo = dict()
-        for userinfo in self.candy_json['records']:
-            userid = userinfo['userid']
-            self.get_candyinfo[userid] = userinfo
+            candy_data = json.load(candy)
+
+        # The rank data
+        self.candy_records = candy_data.get("records") or dict()
```

Comment:
the get() method can take a 2nd argument for a default value.
```suggestion
        self.candy_records = candy_data.get("records", {})
```